src/sync/stream.ts

```diff
@@ -307,27 +485,98 @@ export class StreamManager extends AbortController {
     });
   }
 
-  addTx(tx: IHistoryTx) {
+  /**
+   * This controls the ACK strategy.
+   * @returns {boolean} if we should send an ack message to the server.
+   */
+  shouldACK(): boolean {
+    return (
+      !this.hasReceivedEndStream &&
+      !this.signal.aborted &&
+      this.lastAcked <= this.lastProcSeq &&
+      this.itemQueue.size() <= MIN_QUEUE_SIZE_FOR_ACK
+    );
+  }
+
+  /**
+   * Ack the stream messages.
+   */
+  ack() {
```

Comment:
I'd rather have two method `ackIfNeeded()` and `ack()`. The second one would simply send the ack unconditionally.

Reply:
But then we would never call the `ack`, only the `ackIfNeeded` since we need to check if we are ready to send a message to the server with the ACK.

In this separation we have the `shouldAck` a method that controls the ACK strategy and the `ack` which is called every time we process an event.
The `ack` just checks with the strategy if we should send the message.
What do you think?

Reply:
Ok. We can keep it this way. Thanks for the explanation.